sdk/dotnet/cmd/pulumi-language-dotnet/main.go

```diff
@@ -570,6 +572,10 @@ func (host *dotnetLanguageHost) Run(ctx context.Context, req *pulumirpc.RunReque
 		// Run from source.
 		args = append(args, "run")
 
+		if host.dotnetBuildSucceeded {
```

Comment:
Nit: It might be worth a comment here on _why_ we do this (i.e. if we've already done a build via `GetRequiredPlugins`, then passing `--no-build` here yields around a 1 second savings).